File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n = nums.size();
        int start = 0;
        int end = n -1 ;

        while(start < end) {
            int total = nums[start] + nums[end];
            if( total == target) {
                return {start+1,end+1};
            }
            if(total<target) {
                start++ ;
            }
            if (total>target) {
                end--;
            }
        }   
        return {0,0};
    }
};

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int n = nums.length;
        int i = 0;
        int j = n-1;

        while(i<=j) {
            int total = nums[i] + nums[j];

            if(total==target) {
                return new int[] { i+1 , j+1 };
            }
            else if(total<target) {
                i++ ;
            }
            else {
                j-- ;
            }
        }
        return new int[] {};
    }
}